src/compiler/checker.ts

```diff
@@ -5962,7 +5964,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
     function typeToString(type: Type, enclosingDeclaration?: Node, flags: TypeFormatFlags = TypeFormatFlags.AllowUniqueESSymbolType | TypeFormatFlags.UseAliasDefinedOutsideCurrentScope, writer: EmitTextWriter = createTextWriter("")): string {
         const noTruncation = compilerOptions.noErrorTruncation || flags & TypeFormatFlags.NoTruncation;
-        const typeNode = nodeBuilder.typeToTypeNode(type, enclosingDeclaration, toNodeBuilderFlags(flags) | NodeBuilderFlags.IgnoreErrors | (noTruncation ? NodeBuilderFlags.NoTruncation : 0));
+        const typeNode = nodeBuilder.typeToTypeNode(type, enclosingDeclaration, toNodeBuilderFlags(flags) | NodeBuilderFlags.IgnoreErrors | (noTruncation ? NodeBuilderFlags.NoTruncation : NodeBuilderFlags.None), noTruncation ? InternalNodeBuilderFlags.AllowUnresolvedNames : InternalNodeBuilderFlags.None);

```

Comment:
This scenario seems weird to me as it depends on `NoTruncation` being set in `TypeFormatFlags`. Nonetheless, this is the existing behavior, so I decided to change it but let me know if you disagree and I can rollback.

Reply:
Hm, I think it's a little weird, though I guess it's also true that nobody would have ever _intentionally_ set `AllowUnresolvedNames` here because that was never a valid `TypeFormatFlag`.

Sort of leaning towards _not_ doing this one and seeing what happens? I assume no test changes here either.

Reply:
They don't change. Actually, the latest commit doesn't change any of the tests. I'll rollback this line.